tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,249 @@
+# parameter to determine the length of the animation. The *interval* parameter
+# is used to determine time in milliseconds between drawing of two frames.
+# We will now look at some examples of using
+# :class:`~matplotlib.animation.FuncAnimation` with different artists.
+
+###############################################################################
+# Animating :class:`~matplotlib.lines.Line2D`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# `.pyplot.plot` returns a :class:`~matplotlib.lines.Line2D` collection. The
+# data on this collection can be modified by using the
+# `.lines.Line2D.set_data` function. Therefore we can use this to modify the
+# plot using the function for every frame.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+
+xdata, ydata = [], []
+(line,) = ax.plot(xdata, ydata, c="b")
+ax.grid()
+
+
+def init():
+    ax.set_ylim(-1.1, 1.1)
+    ax.set_xlim(0, 5)
+    del xdata[:]
+    del ydata[:]
+    line.set_data(xdata, ydata)
+    return line,
+
+
+def update(frame):
+    # .set_data resets all the data for the line, so we add the new point to
+    # the existing line data and set that again.
+    xdata.append(frame / 30)
+    ydata.append(np.sin(2 * np.pi * frame / 30))
+
+    line.set_data(xdata, ydata)
+    return (line,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update,
+                              interval=30, init_func=init)
+plt.show()
+
+###############################################################################
+# Animating :class:`~matplotlib.collections.PathCollection`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# `.pyplot.scatter` returns a :class:`~matplotlib.collections.PathCollection`
+# that can similarly be modified by using the
+# `.collections.PathCollection.set_offsets` function.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+t = np.linspace(-4, 4, 400)
+a, b = 3, 2
+delta = np.pi / 2
+
+scat = ax.scatter(np.sin(a * t[0] + delta), np.sin(b * t[0]), c="b", s=2)
+ax.grid()
+ax.set_xlim(-1, 1)
+ax.set_ylim(-1, 1)
+
+
+def update(frame):
+    # .set_offsets also resets the entire data for the collection.
+    # Therefore, we create the entire data in each frame to draw
+    x = np.sin(a * t[:frame] + delta)
+    y = np.sin(b * t[:frame])
+    data = np.stack([x, y]).T
+    scat.set_offsets(data)
+    return (scat,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=400, interval=30)
+plt.show()
+
+
+###############################################################################
+# Animating :class:`~matplotlib.image.AxesImage`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# When we plot an image using `.pyplot.imshow`, it returns an
+# :class:`~matplotlib.image.AxesImage` object. The data in this object can also
+# similarly be modified by using the `.image.AxesImage.set_data` method.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+
+aximg = ax.imshow(rng.uniform(low=0, high=1, size=(10, 10)), cmap="Blues")
```

Comment:
While sorting would be a good example, I think it'll add code to the example that would distract from the process of animating. Animating a rotation might be a better choice. I'll take a look at that.

Reply:
I've been thinking about this, your scatter comment below, and changing `ArtistAnimation` example to show a previous example in a different way and I feel that a new example for images would be a good candidate to show ArtistAnimation too, instead of using a scatter plot for it. That would be closer to a more real application I feel. What do you think? I'm going to add one in a few minutes so that you can see what I mean.